aws-rds-globalcluster/src/main/java/software/amazon/rds/globalcluster/ReadHandler.java

```diff
@@ -22,9 +26,34 @@
 
                     final GlobalCluster targetGlobalCluster = describeGlobalClustersResponse.globalClusters().stream().findFirst().get();
 
-                    return ProgressEvent.defaultSuccessHandler(ResourceModel.builder()
-                            .globalClusterIdentifier(targetGlobalCluster.globalClusterIdentifier())
-                            .build());
+                    return ProgressEvent.defaultSuccessHandler(toResourceModel(targetGlobalCluster));
                 });
     }
+
+    @VisibleForTesting
+    ResourceModel toResourceModel(GlobalCluster cluster) {
+
+        ResourceModel.ResourceModelBuilder builder = ResourceModel.builder();
+
+        builder.globalClusterIdentifier(cluster.globalClusterIdentifier());
+        builder.engine(cluster.engine());
+        builder.engineVersion(cluster.engineVersion());
+        builder.storageEncrypted(cluster.storageEncrypted());
+        builder.deletionProtection(cluster.deletionProtection());
+
+        if (cluster.hasGlobalClusterMembers()) {
+
+            List<GlobalClusterMember> globalClusterMembers = cluster.globalClusterMembers();
+
+            for (GlobalClusterMember globalClusterMember : globalClusterMembers) {
+                if (globalClusterMember.isWriter()) {
+                    builder.sourceDBClusterIdentifier(globalClusterMember.dbClusterArn());
+
+                    break;
+                }
```

Comment:
curious if there is only one writer? if there are more than one how would we determine which one is source?

Reply:
@pabert80 @Dmytrok-amz could you guys confirm that?

Reply:
There is only one writer in Global Databases at the time